controller/api/destination/watcher/endpoints_watcher.go

```diff
@@ -483,6 +602,79 @@ func metricLabels(endpoints *corev1.Endpoints) map[string]string {
 	return labels
 }
 
+func (pp *portPublisher) endpointSliceToAddresses(es *dv1beta1.EndpointSlice) AddressSet {
+
+	addresses := make(map[ID]Address)
+	resolvedPort := pp.resolveESTargetPort(es.Ports)
+	for _, endpoint := range es.Endpoints {
+		if pp.hostname != "" && pp.hostname != *endpoint.Hostname {
+			continue
+		}
+
+		// Without this we get `Add:` twice, once when !ready
+		// and once when ready
+		if !*endpoint.Conditions.Ready {
+			continue
+		}
+
+		// Every endpoint has addresses that correspond to the EndpointSlice addressType field
+		// Consumers must handle different types of addresses
+		// An "addresses" set must contain at least 1 but no more than 100 addresses
+		if endpoint.TargetRef == nil {
+			serviceName := es.Labels[dv1beta1.LabelServiceName]
```

Comment:
What happens if this name is empty? Are we guaranteed that there will be a value for this label?